Reenable cppcheck

I recently had to disable `cppcheck` (see f3b45e111), with apparently same issue as in https://serverfault.com/questions/1052963/pacman-doesnt-work-in-docker-image. At the time of writing I could not find a solution. If this is not resolved, we may need to move away from archlinux for this CI job?

Here is an example of the problem, prior to disabling it:
```
2021-02-10T09:37:20.5496347Z ##[section]Starting: Run cppcheck
2021-02-10T09:37:20.5505177Z ==============================================================================
2021-02-10T09:37:20.5505581Z Task         : Bash
2021-02-10T09:37:20.5505874Z Description  : Run a Bash script on macOS, Linux, or Windows
2021-02-10T09:37:20.5506210Z Version      : 3.182.0
2021-02-10T09:37:20.5506465Z Author       : Microsoft Corporation
2021-02-10T09:37:20.5506893Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/bash
2021-02-10T09:37:20.5507291Z ==============================================================================
2021-02-10T09:37:20.6989049Z Generating script.
2021-02-10T09:37:20.7016624Z ========================== Starting Command Output ===========================
2021-02-10T09:37:20.7033740Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/4b810c6b-40aa-4a3f-832a-1549d4e081ac.sh
2021-02-10T09:37:20.7084943Z + docker build ./tools -f ./tools/cppcheck.Dockerfile -t scipp_cppcheck
2021-02-10T09:37:21.5154822Z Sending build context to Docker daemon  16.38kB
2021-02-10T09:37:21.5155335Z 
2021-02-10T09:37:21.5778738Z Step 1/4 : FROM archlinux
2021-02-10T09:37:22.8330313Z latest: Pulling from library/archlinux
2021-02-10T09:37:23.1572385Z 3a05001ad924: Pulling fs layer
2021-02-10T09:37:23.1573671Z 0a1ff60e1cc4: Pulling fs layer
2021-02-10T09:37:23.4606121Z 0a1ff60e1cc4: Verifying Checksum
2021-02-10T09:37:23.4606583Z 0a1ff60e1cc4: Download complete
2021-02-10T09:37:24.3522539Z 3a05001ad924: Verifying Checksum
2021-02-10T09:37:24.3524698Z 3a05001ad924: Download complete
2021-02-10T09:37:29.4664042Z 3a05001ad924: Pull complete
2021-02-10T09:37:29.5280927Z 0a1ff60e1cc4: Pull complete
2021-02-10T09:37:29.5326853Z Digest: sha256:dcde75fac283b82760b5b64eba36447ced63b2cad28a3ed7abad6c2a72385b23
2021-02-10T09:37:29.5344375Z Status: Downloaded newer image for archlinux:latest
2021-02-10T09:37:29.5373972Z  ---> a8a9773ce02f
2021-02-10T09:37:29.5374659Z Step 2/4 : RUN pacman -Syu --noconfirm cppcheck &&     mkdir -p '/data/'
2021-02-10T09:37:36.3085464Z  ---> Running in f48eea030867
2021-02-10T09:37:36.6774335Z [91merror: failed to initialize alpm library
2021-02-10T09:37:36.6774876Z (could not find or read directory: /var/lib/pacman/)
2021-02-10T09:37:36.8079265Z [0mThe command '/bin/sh -c pacman -Syu --noconfirm cppcheck &&     mkdir -p '/data/'' returned a non-zero code: 255
2021-02-10T09:37:36.8190910Z ##[error]Bash exited with code '255'.
2021-02-10T09:37:36.8230433Z ##[section]Finishing: Run cppcheck
```

--- a/.azure-pipelines/templates/code_quality_checks.yml
+++ b/.azure-pipelines/templates/code_quality_checks.yml
@@ -10,9 +10,6 @@ stages:
         steps:
           - bash: |
               set -ex
-              wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
-              sudo add-apt-repository --yes 'deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-10 main'
-              sudo apt-get update
               sudo apt-get install --yes clang-format-10
             displayName: 'Install tooling'
           - bash: |

--- a/.azure-pipelines/templates/code_quality_checks.yml
+++ b/.azure-pipelines/templates/code_quality_checks.yml
@@ -6,7 +6,7 @@ stages:
       - job: 'cpp_checks'
         displayName: 'C++ Checks'
         pool:
-          vmImage: 'ubuntu-18.04'
+          vmImage: 'ubuntu-20.04'
         steps:
           - bash: |
               set -ex
@@ -27,8 +27,8 @@ stages:
             displayName: 'Run clang-format'
           - bash: |
               set -ex
-              #docker build ./tools -f ./tools/cppcheck.Dockerfile -t scipp_cppcheck
-              #docker run --rm -v "$PWD":/data scipp_cppcheck
+              docker build ./tools -f ./tools/cppcheck.Dockerfile -t scipp_cppcheck
+              docker run --rm -v "$PWD":/data scipp_cppcheck
             displayName: 'Run cppcheck'
 
       - job: 'cmake_checks'

--- a/.azure-pipelines/templates/documentation_build.yml
+++ b/.azure-pipelines/templates/documentation_build.yml
@@ -2,7 +2,7 @@ jobs:
   - job: 'documentation'
     displayName: 'Documentation'
     pool:
-      vmImage: 'ubuntu-18.04'
+      vmImage: 'ubuntu-20.04'
     variables:
       packages_dir: '$(Build.StagingDirectory)/packages'
       docs_build_dir: '$(Build.StagingDirectory)/docs_build'

--- a/.azure-pipelines/templates/test_can_install_and_import.yml
+++ b/.azure-pipelines/templates/test_can_install_and_import.yml
@@ -3,7 +3,7 @@ jobs:
     displayName: 'Can Install and Import - Linux'
     timeoutInMinutes: 10
     pool:
-      vmImage: 'ubuntu-18.04'
+      vmImage: 'ubuntu-20.04'
     variables:
       packages_dir: '$(Build.StagingDirectory)/packages'
     steps:

